codeclimate-wrapper

```diff
@@ -28,35 +51,38 @@ invalid_docker_host() {
   invalid_setup "invalid DOCKER_HOST=$host, must be unset or unix:///var/run/docker.sock"
 }
 
-if [ -n "$DOCKER_MACHINE_NAME" ] && command -v docker-machine > /dev/null 2>&1; then
+if modern_docker_version; then
+  socket_exists || socket_missing
+elif [ -n "$DOCKER_MACHINE_NAME" ] && command -v docker-machine > /dev/null 2>&1; then
   docker-machine ssh $DOCKER_MACHINE_NAME -- \
-    test -S /var/run/docker.sock > /dev/null 2>&1 || socket_missing
+    socket_exists || socket_missing
 
   docker-machine ssh $DOCKER_MACHINE_NAME -- \
     'test -n "$DOCKER_HOST" -a "$DOCKER_HOST" != "unix:///var/run/docker.sock"' > /dev/null 2>&1 \
     && invalid_docker_host $(docker-machine ssh $DOCKER_MACHINE_NAME -- 'echo "$DOCKER_HOST"')
 elif command -v boot2docker > /dev/null 2>&1; then
   boot2docker ssh -- \
-    test -S /var/run/docker.sock > /dev/null 2>&1 || socket_missing
+    socket_exists || socket_missing
 
   boot2docker ssh -- \
     'test -n "$DOCKER_HOST" -a "$DOCKER_HOST" != "unix:///var/run/docker.sock"' > /dev/null 2>&1 \
     && invalid_docker_host $(boot2docker ssh -- 'echo "$DOCKER_HOST"')
 else
-  test -S /var/run/docker.sock || socket_missing
+  socket_exists || socket_missing
   test -n "$DOCKER_HOST" -a "$DOCKER_HOST" != "unix:///var/run/docker.sock" \
     && invalid_docker_host "$DOCKER_HOST"
 fi
 
 docker_run() {
+  local socket_path=$(socket_path)
```

Comment:
I think there are some subtle bugs in the docker-machine case since this invocation of `socket_path` will run on the local system, not the docker-machine VM, so it won't find either socket.

What do you think about restructuring your change like this:

```sh
# Fix our bug, stop hardcoding a literal ;)
socket_path=/var/run/docker.sock

# Modify it for your case
#
# Note: the predicate has to be smarter than just "modern_docker_version". For example,
# I run 1.11 on Linux and the socket is in the default location
if docker_for_mac
  socket_path=/var/tmp/docker.sock
fi

# Now, the rest of the script is exactly as it was on master, except that anywhere
# we've hardcoded the socket path, use $socket_path.
```